### PR TITLE
controld: improve DLM check with stateful_merge_wait

### PR DESCRIPTION
**Pull request by oalbrigt**

### CODE REVIEW

File: extra/resources/controld [hunk 1]
```diff
@@ -228,8 +228,12 @@ controld_monitor() {
...
 
     case $rc in
-      0) dlm_tool ls | grep -q "wait fencing"
-         if [ $? -eq 0 ]; then
+      0) smw=$(dlm_tool status -v | grep "stateful_merge_wait=" | cut -d= -f2)
+         if [ $smw -eq 1 ]; then
+             ocf_log err "DLM status is: stateful_merge_wait"
+             rc=$OCF_ERR_GENERIC
+         elif [ -z "$smw" ] && dlm_tool ls | grep -q "wait fencing" && \
+	      ! stonith_admin -H -V | grep -q "wishes to"; then
```
Comment (gao-yan) [inline extra/resources/controld]: It doesn't work correctly though. "stonith_admin -H" requires a nodename to be supplied. I think it takes "-V" as the nodename here.

File: extra/resources/controld [hunk 2]
```diff
@@ -228,8 +228,12 @@ controld_monitor() {
     killall -0 ${OCF_RESKEY_daemon} >/dev/null 2>&1 ; rc=$?
 
     case $rc in
-      0) dlm_tool ls | grep -q "wait fencing"
-         if [ $? -eq 0 ]; then
+      0) smw=$(dlm_tool status -v | grep "stateful_merge_wait=" | cut -d= -f2)
+         if [ $smw -eq 1 ]; then
```
Comment (gao-yan) [inline extra/resources/controld]: There's error "[: -eq: unary operator expected" if stateful_merge_wait is not available,

Comment (oalbrigt) [inline extra/resources/controld]: Thanks for pointing this out.

File: extra/resources/controld (hunk 1 above)
Comment (oalbrigt) [inline extra/resources/controld]: "-H, --history=value	Retrieve last fencing operation", so I guess I'll have to add a 1. I'll do some testing and fix it.

File: extra/resources/controld (hunk 2 above)
Comment (oalbrigt) [inline extra/resources/controld]: Fixed in: https://github.com/ClusterLabs/pacemaker/pull/1033

File: extra/resources/controld (hunk 1 above)
Comment (kgaillot) [inline extra/resources/controld]: I've added the ability to do stonith_admin -H '*' -V to get the history for all nodes.

Comment (oalbrigt) [inline extra/resources/controld]: Fixed in: https://github.com/ClusterLabs/pacemaker/pull/1035

**Pull request merged**